### PR TITLE
SFR-671 create tests

### PR DESCRIPTION
**Pull request by hokei**
[Project Reno Jira Ticket](http://jira.nypl.org/browse/RENO-XXX)

[ResearchNow Jira Ticket](http://jira.nypl.org/browse/SFR-671)

## **This PR does the following:**
- Creates unit tests for twig templates
- Runs them using a [puppeteer](https://github.com/GoogleChrome/puppeteer) 
- Reconfigures the `bem` function such that it is not reliant on drupal

### UAT Review:
- [ ] Navigate to [the page in Drupal](https://reno-XXX-nypl.pantheonsite.io/)

### Front End Review:
- [ ] View [the example in Pattern Lab](https://reno-XXX-nypl.pantheonsite.io/themes/custom/nypl_emulsify/pattern-lab/public)
- [ ] Check against the [Metronome documentation](http://themetronome.co/components/xxx/?fresh=true)

### CODE REVIEW

File: src/twig/_twig-components/functions/bem.function.php [hunk 1]
```diff
@@ -4,73 +4,44 @@
...
-    $attributes = new Attribute();
-
-    // Iterate the attributes available in context.
-    foreach($context['attributes'] as $key => $value) {
-      // If there are classes, add them to the classes array.
-      if ($key === 'class') {
-        foreach ($value as $class) {
-          $classes[] = $class;
-        }
+function bem(\Twig_Environment &$env, $config) {
```
Comment (hokei) [inline src/twig/_twig-components/functions/bem.function.php:7]: This looks to be re-generated from patternlab and I should find a way to make sure this used going forward

File: src/twig/package.json [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "mocha __tests__/*.js __tests__/**/*.js"
   },
   "keywords": [],
   "author": "",
-  "license": "Apache"
+  "license": "Apache",
```
Comment (helenvholmes) [inline src/twig/package.json]: Is it `Apache` or `Apache 2.0`? I feel like we've been swapping back and forth.

File: src/twig/_twig-components/functions/bem.function.php (hunk 1 above)
Comment (helenvholmes) [reply src/twig/_twig-components/functions/bem.function.php:7]: This entire function file, or just certain lines of the file?

**Review by helenvholmes — APPROVED** [review]
Looks good to me! 👍🏻

File: src/twig/package.json (hunk 2 above)
Comment (hokei) [reply src/twig/package.json]: It throws me a warn when I use "Apache" and so I think it should be "Apache 2.0"

**Comment by helenvholmes** [comment]
Just to be clear, are we blocking this on the `package-lock.json` issue?

**Comment by hokei** [comment]
This is blocked on preventing the automatic replacement of `bem.function.php`
Unblocking it hasn't been my highest priority, but it would entail going in to the emulsify scripts and disabling the scripts that generate the function

It might also be unblocked by the twig->storybook work.

File: patternlab/scripts/twig_functions.sh [hunk 3]
```diff
@@ -25,7 +25,6 @@ then
   # Array of twig functions to copy, starting from the vendor directory
   twig_functions=(
     "drupal-pattern-lab/add-attributes-twig-extension/add_attributes.function.php"
-    "drupal-pattern-lab/bem-twig-extension/bem.function.php"
```
Comment (hokei) [inline patternlab/scripts/twig_functions.sh:28]: This is the main thing that changed

Comment (helenvholmes) [reply patternlab/scripts/twig_functions.sh:28]: Ah, so just pulling this up out of the library code and into our code?

Comment (hokei) [reply patternlab/scripts/twig_functions.sh:28]: Pretty much!  We're manually keeping it in the `src/twig` npm module so it's getting checked in, there's no need to replace it.

Comment (helenvholmes) [reply patternlab/scripts/twig_functions.sh:28]: I think we'll still potentially need it with the storybook/twig changes to interpolate the bem function in all of those files, but we should either include it as a dependency or not have some install script moving it around 😅

**Review by helenvholmes — APPROVED** [review]
👍🏻

**Pull request merged**